lib/src/cpp/domain_participant/builtin/discovery_config/DiscoveryConfig.cpp

```diff
@@ -0,0 +1,531 @@
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_initial_announcements_period_sec(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_initial_announcements_period_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_client_announcement_period(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_client_announcement_period_sec(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_client_announcement_period_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_static_edp_xml_config(
+        const std::string& xml_file,
+        const std::string& profile_id)
+{
+    throw Unsupported("Unsupported");
+}
+
+void clear_static_edp_xml_config(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        int32_t index)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_discovery_protocol(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& discovery_protocol)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_ignore_participant_flags(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& ignore_participant_flags)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_edp(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& edp)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_simple_edp_pubwriter_subreader(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubwriter_subreader)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_simple_edp_pubreader_subwriter(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubreader_subwriter)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_sec,
+        const std::string& duration_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_sec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_duration_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& duration_nanosec)
+{
+    throw Unsupported("Unsupported");
+}
+
+void set_lease_announcement(
```

Comment:
This `set` function with both `sec` and `nanosec` arguments is no longer required, we changed it into two individual set functions for `sec` and `nanosec`